### DISCUSSION TITLE
自定义校验规则不能用 x-validator: "custom" 的方式来使用？

### DISCUSSION THREAD

**Discussion by MisicDemone** [alibaba/formily#1977] [Q&A 2.x]
自定义校验规则不能按 x-validator: "custom" 的简写方式来使用，而自定义校验format可以，但是自定义的校验format通常需要配合message字段使用，不像rule会在定义时带上提示信息。

https://codesandbox.io/s/condescending-fermi-lcj05?file=/App.tsx

**Comment by MisicDemone** [comment]
此处还有个附加问题，自定义format能否在创建时指定提示信息

**Comment by janryWang** [comment]
这个属于约定吧，只有正则格式校验是支持字符串指定的，如果自定义校验规则也支持字符串，就很容易打架了

**Reply by MisicDemone** [reply]
那就应该在自定义校验格式时也能指定message才行，不然除了内置的format，基本都不能直接字符串指定了。

**Reply by janryWang** [reply]
自定义校验一般都是在函数内自己返回message了呀，为啥还要支持指定呢？

**Reply by MisicDemone** [reply]
format的自定义不是仅支持正则嘛，在注册自定义format的时候并没有选项可以指定message

**Reply by janryWang** [reply]
```
{
  format:"url",
  message:"xxx"
}
```
这样是可用的
